extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxInsertDialog.js

```diff
@@ -131,7 +115,7 @@ ve.ui.WikiaInfoboxInsertDialog.prototype.getInfoboxTemplates = function () {
 			} );
 		this.gettingTemplateNames = deferred.promise();
 	}
-	return this.gettingTemplateNames;
+	return deferred.promise();
```

Comment:
On the second call to getInfoboxTemplates (so when this.gettingTemplateNames is set) this code will attempt to call method promise() on undefined (the deferred).
Btw. why to make this (return different thing) in the first place?